ras-events.c

```diff
@@ -102,15 +102,39 @@ static int get_debugfs_dir(char *tracing_dir, size_t len)
 	return ENOENT;
 }
 
+static int wait_access(char *path, int ms)
+{
+	int i;
+	for (i = 0; i < ms; i++) {
+		if (access(path, F_OK) == 0)
+			return 0;
+		usleep(1000);
+	}
+
+	log(ALL, LOG_WARNING, "wait_access() failed, %s not created in %d ms\n", path, ms);
+	return -1;
+}
+
 static int open_trace(struct ras_events *ras, char *name, int flags)
 {
+	int ret = 0;
 	char fname[MAX_PATH + 1];
 
 	strcpy(fname, ras->tracing);
 	strcat(fname, "/");
 	strcat(fname, name);
 
-	return open(fname, flags);
+	ret = wait_access(fname, 100);
```

Comment:
Nothing warrants that 100ms is enough. I would setup it a way bigger, like 30 seconds, as having rasdaemon failing means that RAS instrumentation won't be in place.

Granted, on normal situations, this will be solved a way faster, but the server might be on high load. So, a long timeout won't hurt.